File: apps/web/src/types/issue.ts
export interface Label {
  id: string;
  github_id: number | null;
  name: string;
  color: string;
  description: string | null;
}

export interface Comment {
  id: string;
  github_id: number;
  body: string;
  author_username: string | null;
  author_avatar_url: string | null;
  github_created_at: string;
}

export interface IssueScore {
  id: string;
  issue_id: string;
  score: number;
  recommendation: 'Implement' | 'Investigate' | 'Skip';
  rule_breakdown: Record<string, number>;
  created_at: string;
  updated_at: string;
}

export interface IssueAnalysis {
  id: string;
  issue_id: string;
  summary: string;
  difficulty: 'beginner' | 'intermediate' | 'advanced';
  estimated_time_minutes: number;
  risk: 'low' | 'medium' | 'high';
  component: string;
  languages: string[];
  likely_files: string[];
  merge_probability: number;
  ai_confidence: number;
  model_name: string;
  analysis_version: string;
  analyzed_at: string;

  // AI Coach Extensions
  problem_explanation: string;
  implementation_plan: string[];
  required_knowledge: string[];
  prerequisites: string[];
  acceptance_criteria: string[];
  testing_strategy: string;
  possible_challenges: string[];
  estimated_learning_time: number;
  confidence_reasoning: string;
}

export interface Issue {
  id: string;
  github_id: number;
  number: number;
  title: string;
  body: string | null;
  state: 'open' | 'closed';
  author_username: string | null;
  author_avatar_url: string | null;
  html_url: string;
  comments_count: number;
  is_pull_request: boolean;
  repository_id: string;
  labels: Label[];
  score?: IssueScore | null;
  analysis?: IssueAnalysis | null;
  github_created_at: string;
  github_updated_at: string;
  github_closed_at: string | null;
}

export interface IssueDetail extends Issue {
  comments: Comment[];
}

File: apps/api/app/schemas/issue.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.analysis import IssueAnalysisRead
from app.schemas.score import IssueScoreRead


class LabelRead(BaseModel):
    id: uuid.UUID
    github_id: int | None = None
    name: str
    color: str
    description: str | None = None

    model_config = ConfigDict(from_attributes=True)


class CommentRead(BaseModel):
    id: uuid.UUID
    github_id: int
    body: str
    author_username: str | None = None
    author_avatar_url: str | None = None
    github_created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class IssueRead(BaseModel):
    id: uuid.UUID
    github_id: int
    number: int
    title: str
    body: str | None = None
    state: str
    author_username: str | None = None
    author_avatar_url: str | None = None
    html_url: str
    comments_count: int
    is_pull_request: bool
    repository_id: uuid.UUID
    labels: list[LabelRead] = []
    score: IssueScoreRead | None = None
    analysis: IssueAnalysisRead | None = None
    github_created_at: datetime
    github_updated_at: datetime
    github_closed_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class IssueDetailRead(IssueRead):
    comments: list[CommentRead] = []

    model_config = ConfigDict(from_attributes=True)
